src/arrow-sdk.ts

```diff
@@ -190,17 +190,74 @@ export async function getRecommendedStrategies(
+            type: getReadableContractType(firstRecommendedOption.contract_type, firstRecommendedOption.order_type),
+            price: firstRecommendedOption.price,
+            greeks: firstRecommendedOption.greeks
+        }
+
+        if (secondRecommendedOption === undefined) {
+            return [firstOption]
+        }
+
+        const secondOption = {
+            'long_leg': {
                 ticker: ticker,
                 expiration: readableExpiration,
-                strike: option.strike,
-                type: getReadableContractType(option.contract_type, option.order_type),
-                price: option.price,
-                greeks: option.greeks
-            }
-        })
-        return recommendedOptions
+                strike: secondRecommendedOption.long_leg.strike,
+                type: getReadableContractType(secondRecommendedOption.long_leg.contract_type, secondRecommendedOption.long_leg.order_type),
+                price: secondRecommendedOption.long_leg.price,
+                greeks: secondRecommendedOption.long_leg.greeks
+            },
+            'short_leg': {
+                ticker: ticker,
+                expiration: readableExpiration,
+                strike: secondRecommendedOption.short_leg.strike,
+                type: getReadableContractType(secondRecommendedOption.short_leg.contract_type, secondRecommendedOption.short_leg.order_type),
+                price: secondRecommendedOption.short_leg.price,
+                greeks: secondRecommendedOption.short_leg.greeks
+            },
+            'spread_price': secondRecommendedOption['spread_price'],
+            'spread_greek': secondRecommendedOption['spread_greeks']
+        }
+
+        if (thirdRecommendedOption === undefined) {
+            return [firstOption, secondOption]
+        }
+
+        const thirdOption = {
+            'long_leg': {
+                ticker: ticker,
+                expiration: readableExpiration,
+                strike: thirdRecommendedOption.long_leg.strike,
+                type: getReadableContractType(thirdRecommendedOption.long_leg.contract_type, thirdRecommendedOption.long_leg.order_type),
+                price: thirdRecommendedOption.long_leg.price,
+                greeks: thirdRecommendedOption.long_leg.greeks
+            },
+            'short_leg': {
+                ticker: ticker,
+                expiration: readableExpiration,
+                strike: thirdRecommendedOption.short_leg.strike,
+                type: getReadableContractType(thirdRecommendedOption.short_leg.contract_type, thirdRecommendedOption.short_leg.order_type),
+                price: thirdRecommendedOption.short_leg.price,
+                greeks: thirdRecommendedOption.short_leg.greeks
+            },
+            'spread_price': thirdRecommendedOption['spread_price'],
+            'spread_greek': thirdRecommendedOption['spread_greeks']
+        }
+
+        return [firstOption, secondOption, thirdOption]
```

Comment:
I think this is fine...? I kept getting confusing by `firstOption` not being the same thing as the first option contract lol. It just means a choice, not an option transaction. By the way, I want to get your opinion on something: So option contracts can either be long or short puts or calls. Does that mean that we should have an option transaction type that allows us to combine option contracts together?

Reply:
Yeah I like this idea. So it means option transactions (can form various strategies) are made of options which are single leg long / short calls/puts. I think this is more scalable than what we have going on.